Client/src/Components/AttributeFilter/MembershipField.jsx

```diff
@@ -310,38 +318,41 @@ class MembershipTable extends React.PureComponent {
   }
 
   handleUnknownChange(addUnknown) {
-    this.emitChange(this.getValuesForFilter(), addUnknown);
+    this.setSelections(this.getValuesForFilter(), addUnknown);
   }
 
   handleSelectAll() {
-    const allValues = this.getKnownValues();
-
-    const disabledValues = this.props.activeFieldState.summary.valueCounts
-      .filter(entry => entry.filteredCount === 0)
-      .map(entry => entry.value);
-
-    const filterValues = this.getValuesForFilter();
-
-    if (disabledValues.length === 0 && filterValues == null) {
-      this.emitChange(undefined);
-    } else {
-      let rows = this.getRows();
-      if (this.isSearchEnabled()){
-        rows = filterBySearchTerm(rows, this.props.activeFieldState.searchTerm);
+    const allRows = this.getRows();
+    const searchTerm = this.props.activeFieldState.searchTerm;
+
+    if (!searchTerm) {
+      if (allRows.some(row => row.filteredCount > 0 && !this.isItemSelected(row))) {
+        // At least one row isn't selected. Select all rows.
+        this.setSelections(allRows.map(row => row.value));
+      } else {
+        // All rows are selected. Deselect all rows.
+        this.setSelections([]);
       }
-      if (this.isPaginationEnabled()){
-        rows = selectPage(rows, this.props.activeFieldState.currentPage, this.props.activeFieldState.rowsPerPage);
```

Comment:
As mentioned above, I removed this pagination logic. Is this going to break any desired functionality? Tagging @wbazant as it looks like he wrote this